src/output/clustering_variables.h

```diff
@@ -0,0 +1,310 @@
+	}
+};
+class Score: public Variable{
+public:
+	std::string get_name(){
+		return "score";
+	}
+	double get(const Hsp_context& r){
+		return r.score();
+	}
+};
+class Length: public Variable{
+public:
+	std::string get_name(){
+		return "length";
+	}
+	double get(const Hsp_context& r){
+		return r.length();
+	}
+};
+class PercentIdenticalMatches: public Variable{
+public:
+	std::string get_name(){
+		return "pident";
+	}
+	double get(const Hsp_context& r){
+		return (double)r.identities() * 100 / r.length();
+	}
+};
+class NumberIdenticalMatches: public Variable{
+public:
+	std::string get_name(){
+		return "nident";
+	}
+	double get(const Hsp_context& r){
+		return r.identities();
+	}
+};
+class NumberMismatches: public Variable{
+public:
+	std::string get_name(){
+		return "mismatch";
+	}
+	double get(const Hsp_context& r){
+		return r.mismatches();
+	}
+};
+class NumberPositiveMatches: public Variable{
+public:
+	std::string get_name(){
+		return "positive";
+	}
+	double get(const Hsp_context& r){
+		return r.positives();
+	}
+};
+class NumberGapOpenings: public Variable{
+public:
+	std::string get_name(){
+		return "gapopen";
+	}
+	double get(const Hsp_context& r){
+		return r.gap_openings();
+	}
+};
+class NumberGaps: public Variable{
+public:
+	std::string get_name(){
+		return "gaps";
+	}
+	double get(const Hsp_context& r){
+		return r.gaps();
+	}
+};
+class PercentagePositiveMatches: public Variable{
+public:
+	std::string get_name(){
+		return "ppos";
+	}
+	double get(const Hsp_context& r){
+		return (double)r.positives() * 100.0 / r.length();
+	}
+};
+class QueryFrame: public Variable{
+public:
+	std::string get_name(){
+		return "qframe";
+	}
+	double get(const Hsp_context& r){
+		return r.blast_query_frame();
+	}
+};
+class QueryCoveragePerHsp: public Variable{
+public:
+	std::string get_name(){
+		return "qcovhsp";
+	}
+	double get(const Hsp_context& r){
+		return (double)r.query_source_range().length()*100.0 / r.query.source().length();
+	}
+};
+class SwDiff: public Variable{
+public:
+	std::string get_name(){
+		return "swdiff";
+	}
+	double get(const Hsp_context& r){
+		return r.sw_score() - r.bit_score();
+	}
+};
+class Time: public Variable{
+public:
+	std::string get_name(){
+		return "time";
+	}
+	double get(const Hsp_context& r){
+		return r.time();
+	}
+};
+class SubjectCoveragePerHsp: public Variable{
+public:
+	std::string get_name(){
+		return "scovhsp";
+	}
+	double get(const Hsp_context& r){
+		return (double)r.subject_range().length() * 100.0 / r.subject_len;
+	}
+};
+class UngappedScore: public Variable{
+public:
+	std::string get_name(){
+		return "ungapped_score";
+	}
+	double get(const Hsp_context& r){
+		return score_matrix.bitscore(r.ungapped_score);
+	}
+};
+
+class VariableRegistry{
+private:
+	VariableRegistry(){};
+	// To include new clustering algorithms add the instantiation here and in the cluster_registry.cpp file. Then add it to the StaticConstructor below
+	static QueryStart queryStart;
```

Comment:
Length of query and subject could be useful.

Reply:
Good catch. I meant to add all of the defined variables in the header.